astropy/coordinates/funcs.py

```diff
@@ -159,9 +159,9 @@ def get_sun(time):
     # convert barycentric velocity to units of c, but keep as array for passing in to erfa
     earth_v /= c.to(u.au/u.d).value
 
-    dsun = np.sqrt(np.sum(earth_p**2, axis=-1, keepdims=True))
+    dsun = np.sqrt(np.sum(earth_p**2, axis=-1))
     invlorentz = (1-np.sum(earth_v**2, axis=-1))**0.5
```

Comment:
@mhvk - Hmm, I can't reproduce your result.  If I do the following:
```python
dsun = np.sqrt(np.sum(earth_p**2, axis=-1, keepdims=True))
invlorentz = (1-np.sum(earth_v**2, axis=-1, keepdims=True))**0.5
properdir = erfa.ab(earth_p/dsun, -earth_v, dsun, invlorentz)
```
``test_regression_4996`` fails...

The problem isn't that the next line fails, but rather that with ``keepdims=True``, ``erfa.ab`` broadcasts across the 1 dimension, so you get something that's (1000,1000,3) instead of (1000,3).  

Put another way, the problem is that *both* `dsun` variants are needed - one with a trailing dimension of (1,), and the other with no trailing dimension.

All that said, investigating this made me realize there *is* a broadcast bug here, which I will push up in a new commit momentarily.